socialhome/streams/app/stores/streamStore.js

```diff
@@ -1,25 +1,31 @@
 import Vue from "vue"
 import Vuex from "vuex"
 import ReconnectingWebSocket from "reconnecting-websocket"
+import _defaults from "lodash/defaults"
+import _get from "lodash/get"
 
 import getState from "streams/app/stores/streamStore.state"
-import {actions, mutations, stateOperations} from "streams/app/stores/streamStore.operations"
+import {actions, mutations, streamStoreOerations} from "streams/app/stores/streamStore.operations"
 
 
 Vue.use(Vuex)
 
-function newinstance(WebSocketImpl = ReconnectingWebSocket) {
+function newStreamStore(options) {
     const state = getState()
-    const store = new Vuex.Store({state, mutations, actions})
+    const opts = _defaults({}, {state, mutations, actions}, options)
+    const WebSocketImpl = _get(opts, ["WebSocketImpl"], ReconnectingWebSocket)
+    delete opts.WebSocketImpl
```

Comment:
Test purposes. It has no other use whatsoever. I didn't find a correct way to spy on websocket implementation except by injecting it. After that, I delete it to prevent is from being added to the store's properties. Which is a bit dumb now that I think about it, because it's not used outside tests. I suppose I should add a comment.

Reply:
Generally of course it would be nice to avoid code that exists only for tests, but if you didn't find any other way to do it then a clear comment please and let's live with it - can always fix it if another way comes up 👍